packages/dashboard/e2e/tests/utils/grid.ts

```diff
@@ -0,0 +1,134 @@
+import { Locator, Page } from '@playwright/test';
+import last from 'lodash/last';
+
+import { DragPosition, dragAndDrop } from './dragAndDrop';
+import { center } from './mousePosition';
+import { BoundingBox, getBoundingBox } from './locator';
+
+export const GRID_SIZE = 10;
+
+const WidgetSelector = '[data-gesture=widget]';
+const SelectionSelector = '[data-gesture=selection]';
+const GridSelector = '#container';
+
+type ComponentLibrarySelector = 'kpi' | 'text';
+
+const WidgetSelectorMap: { [key in ComponentLibrarySelector]: string } = {
+  kpi: 'add KPI widget',
+  text: 'add Text widget',
+};
+
+type SelectionAnchor = 'bottom-right';
+
+const SelectionAnchorMap: { [key in SelectionAnchor]: string } = {
+  'bottom-right': '[data-anchor=bottom-right]',
+};
+
+export const gridUtil = (page: Page) => {
+  const dragGenerator = dragAndDrop(page);
+
+  const gridArea = page.locator(GridSelector);
+
+  return {
+    /**
+     *
+     * @param col column of grid to select
+     * @param row row of grid to select
+     * @returns the real x, y postion in the viewport of that cell
+     */
+    cellLocation: async (col: number, row: number): Promise<{ x: number; y: number }> => {
```

Comment:
Consider avoiding declaring return type type annotations. They typically can be inferred by the compiler. Comment applies to most of the return type type annotations of the change.